session/filetransfer.go

```diff
@@ -393,6 +393,18 @@ func (h *FileTransferHandler) InitFileUpload(msg *ws.ProtoMsg, w ResponseWriter)
 		return filetransfer.ErrTxBytesLimitExhausted
 	}
 
+	if info, errStat := os.Stat(*params.Path); errStat != nil {
+		if !os.IsNotExist(errStat) {
+			err = errors.Wrap(errStat, "error checking file location")
+			return err
+		}
+	} else if !info.Mode().IsRegular() {
+		err = errors.New(
+			"conflicting file path: cannot overwrite irregular file",
+		)
+		return err
+	}
+
```

Comment:
we do have [utils/fs.go](https://github.com/mendersoftware/mender-connect/blob/master/utils/fs.go#L41) maybe lets extend that?

Reply:
We need the error returned from stat as well in case the file does not exist.